2333/2/libs/rapper.rb

```diff
@@ -0,0 +1,35 @@
+require 'russian_obscenity'
+
+# class Rapper
+class Rapper
+  attr_reader :name, :battles
+  ROUNDS_ON_BATTLE = 3
+
+  def initialize(name, battles)
+    @name = name
+    @battles = battles
+    RussianObscenity.dictionary = [:default, './my_dictionary.yml']
+  end
+
+  def number_of_bad_words
+    unique_bad_words.map { |word| battles_words.count(word) }.reduce(:+)
```

Comment:
I like how it looks now. But we still have a problem with performance. Imagine we will call this `Rapper#number_of_bad_words` method multiple time. And we do this when we do sorting. On each comparison, we make a call for `RussianObscenity.find` which goes through words and analyze them, then for each bad word we again go through all the words. It should be too long, right?

Please use memoization in order not to recalculate the result of this method execution.

And since we don't mutate the state of our object and we don't pass variables to methods it's fairly simple now ;-).

Please measure the execution time for the current version and then for the version of the app with memoization and share it in the chat with me.